js/libtextsecure.js

```diff
@@ -39276,18 +39298,34 @@ MessageSender.prototype = {
         var myNumber = textsecure.storage.user.getNumber();
         var myDevice = textsecure.storage.user.getDeviceId();
         if (myDevice != 1) {
-            var verified = new textsecure.protobuf.SyncMessage.Verified();
-            verified.state = state;
-            verified.destination = destination;
-            verified.identityKey = identityKey;
+            // First send a null message to mask the sync message.
+            var nullMessage = new textsecure.protobuf.NullMessage();
+
+            // Generate a random int from 1 and 512
+            var buffer = libsignal.crypto.getRandomBytes(1);
+            var paddingLength = (new Uint8Array(buffer)[0] & 0x1ff) + 1;
 
-            var syncMessage = new textsecure.protobuf.SyncMessage();
-            syncMessage.verified = verified;
+            // Generate a random padding buffer of the chosen size
+            nullMessage.padding = libsignal.crypto.getRandomBytes(paddingLength);
 
             var contentMessage = new textsecure.protobuf.Content();
-            contentMessage.syncMessage = syncMessage;
+            contentMessage.nullMessage = nullMessage;
 
-            return this.sendIndividualProto(myNumber, contentMessage, Date.now());
+            return this.sendIndividualProto(destination, contentMessage, Date.now()).then(function() {
```

Comment:
This might be a little easier to read if it were split into two different functions, one for the null message and the other for the verification sync.